### PR TITLE
[IMP] base_report_to_printer: allow connect to protected server

### PR DESCRIPTION
**Pull request by fmdl**
This PR allow to connect to a protected server.

@rousseldenis

### CODE REVIEW

**Comment by fmdl** [comment]
cc @flotho

File: base_report_to_printer/models/printing_server.py [hunk 1]
```diff
@@ -24,6 +24,11 @@ class PrintingServer(models.Model):
         default="localhost", required=True, help="IP address or hostname of the server"
     )
     port = fields.Integer(default=631, required=True, help="Port of the server.")
+    user = fields.Char(help="User name to connect to the server. Empty by default.")
+    password = fields.Char(help="Password to connect to the server. Empty by default.")
+    encryption_policy = fields.Char(
+        help="Encryption Policy to connect to the server. Empty by default."
```
Comment (flotho) [inline base_report_to_printer/models/printing_server.py]: Maybe point out some resources to help user to choose proper values.

File: base_report_to_printer/models/printing_server.py [hunk 2]
```diff
@@ -43,6 +48,13 @@ def _open_connection(self, raise_on_error=False):
             # (see https://github.com/OpenPrinting/pycups/issues/30)
             cups.setServer(self.address)
             cups.setPort(self.port)
+            if self.user:
```
Comment (flotho) [inline base_report_to_printer/models/printing_server.py:64]: you separate 3 cases, could you please explain this reason.
user/password policy should be set up in the same process, you code suppose that cups will allow to connect with a user and no password. Does cups allow this behaviour ?

**Comment by fmdl** [comment]
@sla-subteno-it

**Comment by fmdl** [comment]
@cle-subteno-it

**Review by flotho — APPROVED** [review]

**Review by sla-subteno-it — APPROVED** [review]

**Comment by fmdl** [comment]
@flotho can you merge ?

**Comment by flotho** [comment]
Hi, @fmdl ,

I would be honored but ASFAIK, I have no right to do this. I'm not part of the maintainers of this repo.
Try to ping one of the maintainers https://odoo-community.org/psc-teams/reporting-engine-112

Regarsd

**Comment by fmdl** [comment]
ping @pedrobaeza

**Comment by pedrobaeza** [comment]
Merging according reviews:

/ocabot merge major

**Pull request merged**